keras_nlp/models/gpt2/gpt2_preprocessor.py

```diff
@@ -63,68 +62,65 @@ class GPT2Preprocessor(Preprocessor):
         add_end_token: If true, the preprocessor will append the tokenizer
             end token to each input sequence.
 
+    Call arguments:
+        x: A string `tf.Tensor` or list of python strings.
+        y: Any label data. Will be passed through unaltered.
+        sample_weight: Any label weight data. Will be passed through unaltered.
+        sequence_length: Pass to override the configured `sequence_length` of
+            the layer.
+        add_start_token: Pass to override the configure value of
```

Comment:
(cannot remember if I have posted this or not...)

So in call and __init__, we both have `add_start_token`, but they are actually different types. This could lead to some confusion.

Reply:
There was a comment chain above that confused me, but basically if we go with this override approach, we need three possible types in call so it can act as an actual override. `None` default use configured value, `True` override to true, `False` override to false. If you only accept True/False here, you have made the init argument obsolete right?